File: exercism/python/isbn-verifier/isbn_verifier.py
def is_valid(isbn: str) -> bool:
    sanitized_input: str = isbn.replace("-", "")

    if len(sanitized_input) != 10:
        return False

    digit_sum: int = 0
    for i in range(0, 10):
        curr: str = sanitized_input[i]
        if i < 9 and curr.isdigit():
            digit_sum += int(curr) * (10 - i)
        else:
            if curr.isdigit():
                digit_sum += int(curr)
            elif curr == "X":
                digit_sum += 10
            else:
                return False

    return digit_sum % 11 == 0

File: exercism/typescript/isbn-verifier/isbn-verifier.ts
export function isValid(isbn: string): boolean {
  const cleanedIsbn = isbn.replace(/-/g, '');
  const regex = /^[0-9]{9}[0-9X]$/;

  if (!regex.test(cleanedIsbn)) {
    return false;
  }

  let sum = 0;
  for (let i = 0; i < cleanedIsbn.length; i++) {
    const char = cleanedIsbn[i];
    const value = char === 'X' ? 10 : parseInt(char, 10);
    sum += value * (10 - i);
  }

  return sum % 11 === 0;
}
